### PR TITLE
feat(authorisation_model): adds roles to the authorisation model

### PR DESCRIPTION
**Pull request by ale8k**
## Description

Adds roles to our authorisation model. I also opted to make a make-target, and some doccos on how to do this.

## Engineering checklist
*Check only items that apply*

- [x] Documentation updated
- [x] Covered by unit tests
- [ ] Covered by integration tests

### CODE REVIEW

**Review by SimoneDutto — APPROVED** [review]
lgtm

File: openfga/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# openfga
+
+## Introduction
+The OpenFGA package holds our authorisation model and a go embed. It also holds
```
Comment (kian99) [inline openfga/README.md]:
```suggestion
The OpenFGA package holds our authorisation model and a go embed to pass the auth model into tests.
It also holds
```

File: openfga/tests.fga.yaml [hunk 2]
```diff
@@ -1,12 +1,5 @@
-# OpenFGA CLI instructions
-#
-# Installation: https://github.com/openfga/cli?tab=readme-ov-file#installation
-# Command: fga model test --tests tests.fga.yaml
```
Comment (kian99) [inline openfga/tests.fga.yaml:4]: Let's keep this

File: openfga/tests.fga.yaml [hunk 3]
```diff
@@ -1,12 +1,5 @@
...
-
 model_file: ./authorisation_model.fga
 
-# In order to avoid the potential entanglement of separate tests the tuples are artifically split into groups using this naming convention: (type):(2-letter test name)-(type)-(id)
-# The GitHub action supports running all tests in a directory, but keeping them in a single file improves the local development experience because the CLI does not.
```
Comment (kian99) [inline openfga/tests.fga.yaml:9]: Why not keep this?

File: openfga/tests.fga.yaml (hunk 2 above)
Comment (ale8k) [reply openfga/tests.fga.yaml:4]: I've moved it to the readme, which I think is sufficient.

File: openfga/tests.fga.yaml (hunk 3 above)
Comment (ale8k) [reply openfga/tests.fga.yaml:9]: I've detailed in the README how we do tests, you can see in the CLI when using -h it isn't possible, don't think we need this here.

Comment (kian99) [reply openfga/tests.fga.yaml:9]: I don't see this bit in the readme?

**Review by kian99 — APPROVED** [review]
lgtm bar one comment

**Pull request merged**